File: backend/complex/handle-temporary-contents/internal/server/health/health.go
package health

import (
	"context"

	healthPb "github.com/EchoSkorJjj/IS213-Education-Helper/handle-temporary-contents/pb/health"
)

type Server struct {
	healthPb.UnimplementedHealthServer
}

func NewServer() *Server {
	return &Server{}
}

func (s *Server) Check(ctx context.Context, req *healthPb.HealthCheckRequest) (*healthPb.HealthCheckResponse, error) {
	return &healthPb.HealthCheckResponse{
		Status: healthPb.HealthCheckResponse_SERVING,
	}, nil
}

func (s *Server) Watch(req *healthPb.HealthCheckRequest, srv healthPb.Health_WatchServer) error {
	var response *healthPb.HealthCheckResponse
	response.Status = healthPb.HealthCheckResponse_SERVING
	srv.Send(response)
	return nil
}

File: backend/simple/subscriptions/src/proto-services/health.cpp
#include <grpcpp/grpcpp.h>

#include "../../pb/health.grpc.pb.h"

class HealthService final : public grpc::health::v1::Health::Service {
  grpc::Status Check(grpc::ServerContext *context,
                     const grpc::health::v1::HealthCheckRequest *request,
                     grpc::health::v1::HealthCheckResponse *response) override {
    response->set_status(grpc::health::v1::HealthCheckResponse::SERVING);
    return grpc::Status::OK;
  }

  grpc::Status Watch(grpc::ServerContext *context,
                     const grpc::health::v1::HealthCheckRequest *request,
                     grpc::ServerWriter<grpc::health::v1::HealthCheckResponse>
                         *writer) override {
    grpc::health::v1::HealthCheckResponse response;
    response.set_status(grpc::health::v1::HealthCheckResponse::SERVING);
    writer->Write(response);
    return grpc::Status::OK;
  }
};